service/src/main/java/com/ge/predix/acs/config/CloudRedisConnectionFactoryConfig.java

```diff
@@ -0,0 +1,186 @@
+ * limitations under the License.
+ *******************************************************************************/
+
+package com.ge.predix.acs.config;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cloud.config.java.AbstractCloudConfig;
+import org.springframework.cloud.service.PooledServiceConnectorConfig;
+import org.springframework.cloud.service.PooledServiceConnectorConfig.PoolConfig;
+import org.springframework.cloud.service.common.RedisServiceInfo;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.data.redis.connection.RedisConnectionFactory;
+import org.springframework.data.redis.connection.jedis.JedisConnectionFactory;
+
+import redis.clients.jedis.JedisPoolConfig;
+
+/**
+ * DataSourceConfig used for all cloud profiles.
+ *
+ * @author 212406427
+ */
+@Configuration
+@Profile({ "cloud-redis" })
+public class CloudRedisConnectionFactoryConfig extends AbstractCloudConfig {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CloudRedisConnectionFactoryConfig.class);
+
+    @Value("${DECISION_CACHE_NAME:acs-redis}")
+    private String decisionCacheName;
+    @Value("${DECISION_REDIS_MIN_ACTIVE:0}")
+    private int decisionMinActive;
+    @Value("${DECISION_REDIS_MAX_ACTIVE:100}")
+    private int decisionMaxActive;
+    @Value("${DECISION_REDIS_MAX_WAIT_TIME:2000}")
+    private int decisionMaxWaitTime;
+    @Value("${DECISION_REDIS_SOCKET_TIMEOUT:3000}")
+    private int decisionSoTimeout;
+    @Value("${DECISION_REDIS_USE_JEDIS_FACTORY:false}")
+    private boolean decisionUseJedisFactory;
+
+    @Value("${RESOURCE_CACHE_NAME:acs-resource-redis}")
+    private String resourceCacheName;
+    @Value("${RESOURCE_REDIS_MIN_ACTIVE:0}")
+    private int resourceMinActive;
+    @Value("${RESOURCE_REDIS_MAX_ACTIVE:100}")
+    private int resourceMaxActive;
+    @Value("${RESOURCE_REDIS_MAX_WAIT_TIME:2000}")
+    private int resourceMaxWaitTime;
+    @Value("${RESOURCE_REDIS_SOCKET_TIMEOUT:3000}")
+    private int resourceSoTimeout;
+    @Value("${RESOURCE_REDIS_USE_JEDIS_FACTORY:false}")
+    private boolean resourceUseJedisFactory;
+
+    @Value("${SUBJECT_CACHE_NAME:acs-subject-redis}")
+    private String subjectCacheName;
+    @Value("${SUBJECT_REDIS_MIN_ACTIVE:0}")
+    private int subjectMinActive;
+    @Value("${SUBJECT_REDIS_MAX_ACTIVE:100}")
+    private int subjectMaxActive;
+    @Value("${SUBJECT_REDIS_MAX_WAIT_TIME:2000}")
+    private int subjectMaxWaitTime;
+    @Value("${SUBJECT_REDIS_SOCKET_TIMEOUT:3000}")
+    private int subjectSoTimeout;
+    @Value("${SUBJECT_REDIS_USE_JEDIS_FACTORY:false}")
+    private boolean subjectUseJedisFactory;
+
+    @Bean(name = { "redisConnectionFactory", "decisionRedisConnectionFactory" })
```

Comment:
This looks good , since we are repeating the config variable for redis connections three time, can we create a new class and instaniate three objects of it one each for resource,subject and decision cache.